psse/psse-converter/src/main/java/com/powsybl/psse/converter/PsseImporter.java

```diff
@@ -300,62 +339,67 @@ public Network importData(ReadOnlyDataSource dataSource, NetworkFactory networkF
+                ToDoubleFunction<Object> branchToReactance = branch -> branch instanceof PsseNonTransformerBranch ? ((PsseNonTransformerBranch) branch).getX() : ((PsseTransformer) branch).getSecondRecord().getX12();
+                Predicate<Object> branchToIsTransformer = branch -> branch instanceof PsseTransformer;
+                ContainersMapping containerMapping = ContainersMapping.create(psseModel.getBuses(), branches, PsseBus::getI, branchToNum1,
+                    branchToNum2, branchToResistance, branchToReactance, branchToIsTransformer,
+                    busNums -> "VL" + busNums.iterator().next(), substationNum -> "S" + substationNum++);
+
+                boolean ignoreBaseVoltage = ConversionParameters.readBooleanParameter(FORMAT, parameters, IGNORE_BASE_VOLTAGE_PARAMETER,
+                        ParameterDefaultValueConfig.INSTANCE);
+                PerUnitContext perUnitContext = new PerUnitContext(psseModel.getCaseIdentification().getSbase(), ignoreBaseVoltage);
+
+                // create buses
+                createBuses(psseModel, containerMapping, perUnitContext, network);
+
+                //Create loads
+                for (PsseLoad psseLoad : psseModel.getLoads()) {
+                    createLoad(psseLoad, containerMapping, network);
+                }
 
-            //Create shunts
-            for (PsseFixedShunt psseShunt : psseModel.getFixedShunts()) {
-                createShuntCompensator(psseShunt, perUnitContext, containerMapping, network);
-            }
+                //Create shunts
+                for (PsseFixedShunt psseShunt : psseModel.getFixedShunts()) {
+                    createShuntCompensator(psseShunt, perUnitContext, containerMapping, network);
+                }
 
-            for (PsseGenerator psseGen : psseModel.getGenerators()) {
-                createGenerator(psseGen, containerMapping, network);
-            }
+                for (PsseGenerator psseGen : psseModel.getGenerators()) {
+                    createGenerator(psseGen, containerMapping, network);
+                }
 
-            for (PsseNonTransformerBranch psseLine : psseModel.getNonTransformerBranches()) {
-                createLine(psseLine, containerMapping, perUnitContext, network);
-            }
+                for (PsseNonTransformerBranch psseLine : psseModel.getNonTransformerBranches()) {
+                    createLine(psseLine, containerMapping, perUnitContext, network);
+                }
 
-            for (PsseTransformer psseTfo : psseModel.getTransformers()) {
-                createTransformer(psseTfo, containerMapping, perUnitContext, network);
+                for (PsseTransformer psseTfo : psseModel.getTransformers()) {
+                    createTransformer(psseTfo, containerMapping, perUnitContext, network);
+                }
+
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
             }
```

Comment:
Is this catch OK? There is probably to many try/catch level where only one is sufficient?

Reply:
proposition of modification done